src/services/auth.service.ts

```diff
@@ -46,4 +47,36 @@ export class AuthorizationService {
       return false;
     }
   }
+
+  async currentUserChecker(
+    action: Action
+  ): Promise<ITokenPayload | boolean> {
+    const jwt = Container.get(JWTService);
+    const redis = Container.get(RedisService);
+    try {
+      let token = action.request.headers['authorization'];
+      if (!token) {
+        return false;
+      }
+      if (token.startsWith('Bearer ')) {
+        // Remove Bearer from authentication scheme header
+        token = token.replace('Bearer ', '');
+      }
+      const payload = await jwt.verifyJWT(token);
+      const {
+        data: { email }
+      } = payload;
+      const tokenIsBlacklisted: number = await redis.isMemberOfSet({
+        email,
+        token
+      });
+      if (!!tokenIsBlacklisted) {
+        return false;
+      }
+      return payload;
+    } catch (error) {
+      // Here we should do something with the error like loggin
+      return false;
+    }
```

Comment:
I think all of these things could be done in a valid user middleware  what do you think?